### DISCUSSION TITLE
Export EIP-712 encoding functions

### DISCUSSION THREAD

**Discussion by frangio** [wevm/viem#1416] [Idea / Feature Request]
In [`hashTypedData.ts`](https://github.com/wagmi-dev/viem/blob/0d5cd922e3a63e1173b26b2a20641f8df2273919/src/utils/signature/hashTypedData.ts) there are many functions that can be useful when working with EIP-712 signatures: `hashStruct`, `hashType`, `encodeData`, etc.

It would be great if these were exported.

**Comment by 0xdapper** [comment]
I second this, and this should be relatively simple too. Only time i find myself reaching to ethers is when I need to debug some 712 encoding bug.

**Reply by smartcontrart** [reply]
Same here.

**Reply by Clonescody** [reply]
+1 would be very useful

**Reply by CantaloupeBob** [reply]
Looking for something like this as well

**Reply by lorem-ipsumm** [reply]
I concur
